File: e-shop-back-end/src/main/java/com/commerce/eshop/DTO/montonio/TransactionPayload.java
package com.commerce.eshop.DTO.montonio;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TransactionPayload {
    private String accessKey;
    private String merchantReference;
    private String returnUrl;
    private String notificationUrl;
    private String currency;
    private BigDecimal grandTotal;
    private String locale;
    private BillingAddress billingAddress;
    private ShippingAddress shippingAddress;
    private List<LineItem> lineItems;
    private Payment payment;

    // Getters and setters

    // You can generate getters and setters using your IDE or manually
}

File: e-shop-front-end/e-shop/src/Domain/montonio/forPostTransaction/TransactionPayload.ts
import { ICartItem } from "../../ICartItem";
import { Address } from "./Address";
import { LineItem } from "./LineItem";
import { PaymentMethodTransaction } from "./PaymentMethodTransaction";

export interface TransactionPayload {
    accessKey: string;
    merchantReference: string;
    returnUrl: string;
    notificationUrl: string;
    currency: string;
    grandTotal: number;
    locale: string;
    billingAddress: Address;
    shippingAddress: Address;
    lineItems: LineItem[];
    payment: PaymentMethodTransaction;
}
